hvpy/api.py

```diff
@@ -1,13 +1,76 @@
 __all__ = ["fake_api_call"]
 
+from datetime import datetime
+from typing import Optional
+import requests
+from pydantic import BaseModel, Field
 
-def fake_api_call():
+
+class InputParameters(BaseModel):
+    date: datetime
+    sourceId: int
+    jpip: Optional[bool] = False
+    Json: Optional[bool] = Field(False, alias="json")
+
+
+def parse_response(response: requests.Response, output_parameters: str):
     """
     _summary_
 
+    Parameters
+    ----------
+    response: ?
+        _description_
+    output_parameters: ?
+        _description_
+
     Returns
     -------
     _type_
         _description_
     """
-    return {}
+    if output_parameters == "binary":
+        return response.content
+    elif output_parameters == "url":
+        return response.url
+    elif output_parameters == "json":
+        return response.json()
+
+
+def execute_api_call(url: str, input_parameters: dict, output_parameters: str):
+    """
+    _summary_
+
+    Parameters
+    ----------
+    url: str
+        _description_
```

Comment:
"Endpoint url"

Reply:
Should we keep BASE_URL in .env?

Reply:
We can leave it hard coded as BASE_URL for now. Maybe put it in io.py. It will need to be defined in one place if we want to be able to override it to other mirrors later on.